xmodule/library_content_block.py

```diff
@@ -534,7 +572,7 @@ def studio_post_duplicate(self, store, source_block):
         user_perms = self.runtime.service(self, 'studio_user_permissions')
         if not self.tools:
             raise RuntimeError("Library tools unavailable, duplication will not be sane!")
-        self.tools.update_children(self, user_perms, version=self.source_library_version)
+        self.tools.trigger_update_children_task(self, user_perms, version=self.source_library_version)
 
         self._copy_overrides(store, user_id, source_block, self)
```

Comment:
@connorhaugh I think there's a race-condition bug here. Because `update_children` now happens in a Celery task, `_copy_overrides` is likely to be invoked *before* the children are imported into the duplicate lib content block, which seems to prevent those new children from properly getting their overrides from the original lib content block.

Reproduction:
* in the unit editor in Studio, add a library content block, pointing at any type of library
* click "View All" to see the library content block's children
* override the titles of one of the child blocks. go back to the unit editor.
* on the library content block, hit the dots, hit "duplicate"
* click into "View All" of the new duplicate lib content block.

Expected behavior (confirmed on master):
* the child's overriden title is also overriden in the duplicate lib content block.

Actual behavior (confirmed on this branch):
* the child has the original title from the source library.

I'm thinking about potential fixes currently. I don't

Reply:
I think what I will try to do is to make a new `duplicate_library_content_block_task` Celery task, which calls out to update_children first, and then calls out to _copy_overrides next. All the code could live in `tasks.py` so we don't have circular imports.

Reply:
It is taking me a while and some exploration to understand why this "update children, then copy overrides" operation is necessary.
If, for original block A and duplicate block B, we simply set B's children to be a list of copies of A's children, then we would be sorted, right?
The only reason I can understand why this this problem was solved using update_children is to provide the following expected behavior:

1. Block A has 3 children. You change block 1's name and delete block 2.
2. You copy Block A to make block B.
3. You expect block 2 to be in block B's children.

The other possible reasons this was solved this way are coding choices: we want to maintain as we possible paths to do a thing, these are the tools I already have, etc.

I know your solution (waiting for update_children_task to finish) will work, and if that is easy (which it may not be) that will work, but this is maybe worth exploring.

Reply:
@connorhaugh You know, I honestly don't know why we're importing-then-copying-overrides rather than just copying the blocks!